src/abstractdataframe/iteration.jl

```diff
@@ -20,6 +20,14 @@ end
 Base.summary(dfrs::DataFrameRows) = "$(length(dfrs))-element DataFrameRows"
 Base.summary(io::IO, dfrs::DataFrameRows) = print(io, summary(dfrs))
 
+# Trying to iterate over a DataFrame directly (for row in df) tends to
+# be one of the first things a beginner does. Allowing a MethodError
+# to be raised is more "correct", but an explanation of the
+# appropriate function for iterating rows is more friendly.
+Base.iterate(::AbstractDataFrame) =
+    error("AbstractDataFrame is not iterable. Use eachrow(df) to get a row iterator" *
+          " or eachcol(df) to get a column iterator")
```

Comment:
@nalimilan - are you OK with this?

Reply:
Looks fine to me. :)

Reply:
I use `f() = dostuff()` if `dostuff()` is a single expression even if it spans more than one line (like in the case here). But this is a personal preference, and maybe @nalimilan would disagree 😄.

We use the style:
```
long_function_name_with_many_parameters(x) =
    single_expression_that_does_stuff()
```
a lot in DataFrames.jl (this is especially useful when you have many parameters with long signature).

Reply:
That's fine with me, as long as there's a single expression.